src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWallet.scala

```diff
@@ -1,36 +1,75 @@
 package org.ergoplatform.nodeView.wallet
 
-
-import org.ergoplatform.modifiers.ErgoPersistentModifier
+import akka.actor.{ActorRef, ActorSystem, Props}
+import org.ergoplatform.modifiers.{ErgoFullBlock, ErgoPersistentModifier}
 import org.ergoplatform.modifiers.mempool.ErgoTransaction
+import org.ergoplatform.nodeView.wallet.ErgoWalletActor._
 import org.ergoplatform.settings.ErgoSettings
 import scorex.core.VersionTag
 import scorex.core.transaction.wallet.Vault
 import scorex.core.utils.ScorexLogging
 
-import scala.util.{Success, Try}
+import scala.util.{Failure, Success, Try}
+import org.ergoplatform.local.TransactionGenerator.StartGeneration
+import org.ergoplatform.local.TransactionGeneratorRef
+import org.ergoplatform.nodeView.history.ErgoHistoryReader
+
+
 
-class ErgoWallet extends Vault[ErgoTransaction, ErgoPersistentModifier, ErgoWallet]
-  with ScorexLogging {
+class ErgoWallet(nodeViewHolderRef: ActorRef,
```

Comment:
`nodeViewHolderRef` and `historyReader` looks quite bad here.
`Wallet` should not send messages to `NodeViewHolder`,  `NodeViewHolder` should call `Wallet` when needed.
And interaction between `Wallet` and `History` should be done by `NodeViewHolder`, their direct communication make code tight coupled